Tone/instrument/NoiseSynth.ts

```diff
@@ -0,0 +1,127 @@
+
+import { AmplitudeEnvelope } from "../component/envelope/AmplitudeEnvelope";
+import { NormalRange, Time } from "../core/type/Units";
+import { deepMerge, optionsFromArguments } from "../core/util/Defaults";
+import { RecursivePartial } from "../core/util/Interface";
+import { Noise, NoiseType } from "../source/Noise";
+import { Instrument } from "./Instrument";
+import { Synth, SynthOptions } from "./Synth";
+
+interface NoiseSynthOptions extends SynthOptions {
+	noise: {
+		type: NoiseType,
+	};
+}
+
+/**
+ * Tone.NoiseSynth is composed of a noise generator (Tone.Noise), one filter (Tone.Filter),
+ * and two envelopes (Tone.Envelop). One envelope controls the amplitude
+ * of the noise and the other is controls the cutoff frequency of the filter.
+ * <img src="https://docs.google.com/drawings/d/1rqzuX9rBlhT50MRvD2TKml9bnZhcZmzXF1rf_o7vdnE/pub?w=918&h=242">
+ *
+ * @example
+ * var noiseSynth = new Tone.NoiseSynth().toMaster();
+ * noiseSynth.triggerAttackRelease("8n");
+ */
+export class NoiseSynth extends Instrument<NoiseSynthOptions> {
+
+	readonly name = "NoiseSynth";
+
+	/**
+	 * The noise source.
+	 */
+	readonly noise: Noise;
+
+	/**
+	 * The amplitude envelope.
+	 */
+	readonly envelope: AmplitudeEnvelope;
+
+	constructor(options?: RecursivePartial<NoiseSynthOptions>)
+	constructor() {
+		super(optionsFromArguments(NoiseSynth.getDefaults(), arguments));
+		const options = optionsFromArguments(NoiseSynth.getDefaults(), arguments);
+		this.noise = new Noise(options.noise);
+		this.envelope = new AmplitudeEnvelope(options.envelope);
+
+		// connect the noise to the output
+		this.noise.chain(this.envelope, this.output);
+	}
+
+	static getDefaults(): NoiseSynthOptions {
+		return {
+			...deepMerge(Synth.getDefaults(), {
```

Comment:
Similar to the above comment, this should extend InstrumentOptions and not SynthOptions.